### DISCUSSION TITLE
error finding file when running inference on colab

### DISCUSSION THREAD

**Discussion by akshitasax** [talmolab/sleap#780] [Help!] [answered]
Hi,

I uploaded trained models from the SLEAP GUI onto my google drive to run inference on google colab. I wrote the path to model in the command for running inference on the first 200 frames using the top down pipeline, and I'm pretty sure there's no mistake in the writing of the path. However, upon running, it gives an error that says no such file or directory found. Attached are screenshots of colab and the drive directory where my models are saved. What could be the issue?
<img width="1014" alt="Screen Shot 2022-06-06 at 7 26 39 AM" src="https://user-images.githubusercontent.com/104799583/172181153-a1d4ddb6-05d9-49cd-9f91-ef8effa85d34.png">
<img width="1193" alt="Screen Shot 2022-06-06 at 7 26 18 AM" src="https://user-images.githubusercontent.com/104799583/172181173-ffe7ea5f-be75-4009-b36a-bb9da964b060.png">
<img width="1131" alt="Screen Shot 2022-06-06 at 7 28 32 AM" src="https://user-images.githubusercontent.com/104799583/172181472-78c5f587-081c-416c-a0db-7d0ca2ba24e5.png">

Thank you.

**Comment by roomrys** [comment, marked as answer]
Hi @akshitasax,

Can you share the code you used to mount your drive? It should look similar to:
```py
from google.colab import drive
drive.mount('/content/drive/')
```

And the code where you navigate to your directory:
```py
import os
os.chdir("/content/drive/My Drive/sleap")
!unzip colab.slp.training_job.zip
!ls
```

Thanks,
Liezl

**Reply by akshitasax** [reply]
Hi Liezl,

Here is the code for mounting my drive and where I navigated to my directory. I didn't unzip any file as I didn't use the training job package, I uploaded the entire project directory from my desktop where I have the SLEAP GUI. The trained models are under "models"
<img width="1083" alt="Screen Shot 2022-06-06 at 10 36 06 AM" src="https://user-images.githubusercontent.com/104799583/172214835-456bfac0-6c28-4ff9-9e42-55bede51e35e.png">

Thanks,
Akshita

**Reply by roomrys** [reply]
@akshitasax 

Thanks for the extra info! 

So it looks like you are already in the `My Drive/5-20-2022 CS Day` directory which means you can access the models through `models/<name of model>` instead of `My Drive/5-20-2022 CS Day/models/<name of model>`. 

Just double checking that you have a video named `colab_demo.mp4` in the `My Drive/5-20-2022 CS Day` directory? If not, you should upload the video you want to run inference on (and name it `colab_demo.mp4` or change the name of the video in `sleap-track` command).

After verifying you have the `colab_demo.mp4` video in the `My Drive/5-20-2022 CS Day` directory, try re-running `sleap-train` with the new models path (i.e. removing the `My Drive/5-20-2022 CS Day` from both model paths). Let us know how that works.

```py
!sleap-track colab_demo.mp4 \
    --frames 0-200 \
    --tracking.tracker simple \
    -m models/second220531_154616.centered_instance.n=51 \
    -m models/second220531_154616.centroid.n=51
```

Thanks,
Liezl

**Reply by akshitasax** [reply]
Thank you for that! It worked. One more thing: is there a way to specify settings such as culling to instance count, or connecting single track breaks or selecting a matching method, when running inference on colab? 

Thank you,
Akshita

**Reply by roomrys** [reply]
@akshitasax 

Yes, you can use the [SLEAP CLI Commands](https://sleap.ai/guides/cli.html) in colab (make sure to insert a `!` in front of the CLI command). The command you are looking for is [`sleap-track`](https://sleap.ai/guides/cli.html#sleap-track) which has the following options (and more!):
```
  --tracking.tracker TRACKING.TRACKER
                        Options: simple, flow, None (default: None)
  --tracking.target_instance_count TRACKING.TARGET_INSTANCE_COUNT
                        Target number of instances to track per frame.
                        (default: 0)
  --tracking.pre_cull_to_target TRACKING.PRE_CULL_TO_TARGET
                        If non-zero and target_instance_count is also non-
                        zero, then cull instances over target count per frame
                        *before* tracking. (default: 0)
  --tracking.pre_cull_iou_threshold TRACKING.PRE_CULL_IOU_THRESHOLD
                        If non-zero and pre_cull_to_target also set, then use
                        IOU threshold to remove overlapping instances over
                        count *before* tracking. (default: 0)
  --tracking.post_connect_single_breaks TRACKING.POST_CONNECT_SINGLE_BREAKS
                        If non-zero and target_instance_count is also non-
                        zero, then connect track breaks when exactly one track
                        is lost and exactly one track is spawned in frame.
                        (default: 0)
  --tracking.clean_instance_count TRACKING.CLEAN_INSTANCE_COUNT
                        Target number of instances to clean *after* tracking.
                        (default: 0)
  --tracking.clean_iou_threshold TRACKING.CLEAN_IOU_THRESHOLD
                        IOU to use when culling instances *after* tracking.
                        (default: 0)
```

Let us know if you have any other questions!

Thanks,
Liezl
